addons/product/models/product_template.py

```diff
@@ -441,6 +441,10 @@ def write(self, vals):
                 'image_128',
                 'can_image_1024_be_zoomed',
             ])
+            # touch all products that will fall back on the template field
+            self.product_variant_ids.filtered(
+                lambda product: not product.image_variant_1920
+            ).write({})
```

Comment:
The `filtered` part will probably be too costly for performance, as it will require not only reading the database for potentially many variants, but also the disk because that's how the binary field works.

It's very likely much more efficient in this case to simply write on all variants without distinction (there should be just 1 update query per batch of 1000 variants).